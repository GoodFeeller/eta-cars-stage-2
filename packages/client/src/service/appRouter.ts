import { createTRPCProxyClient, httpBatchLink } from '@trpc/client'
import { appRouterType } from '../../../server/src/appRouter'
const trpc = createTRPCProxyClient<appRouterType>({
  links: [
    httpBatchLink({
      url: 'https://eta-cars-internship-api.onrender.com/trpc',
    }),
  ],
})

export default trpc
