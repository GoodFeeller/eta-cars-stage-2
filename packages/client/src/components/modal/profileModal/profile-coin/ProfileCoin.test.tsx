import { render, screen } from '@testing-library/react'
import { IProfileCoin, IProfileTotal } from '../../../../service/ProfileTypes'
import ProfileCoin from './ProfileCoin.tsx'
import userEvent from '@testing-library/user-event'

const coin: IProfileCoin = {
  id: 'bitcoin',
  totalPrice: 20000,
  count: 3,
  symbol: 'BTC',
}
describe('Profile coin test', () => {
  test('Full page snapshot', () => {
    render(<ProfileCoin coin={coin} />)
    const profile = screen.queryByTestId('profile-coin')
    expect(profile).toMatchSnapshot()
  })
  test('Remove test', () => {
    render(<ProfileCoin coin={coin} />)
    localStorage.clear()
    localStorage.setItem('bitcoin', JSON.stringify(coin))
    localStorage.setItem(
      '//profile//',
      JSON.stringify({
        id: ['bitcoin', 'litecoin'],
        totalPrice: 30000,
      }),
    )
    const btn = screen.queryByText('remove') as HTMLButtonElement
    userEvent.click(btn)
    const bitcoin = localStorage.getItem('bitcoin')
    const profile = localStorage.getItem('//profile//')
    if (profile) {
      const profileObj: IProfileTotal = JSON.parse(profile)
      expect(profileObj.id).toEqual(['litecoin'])
      expect(profileObj.totalPrice).toBe(10000)
    } else expect(false).toBe(true)
    expect(bitcoin).toBeNull()
  })
})
