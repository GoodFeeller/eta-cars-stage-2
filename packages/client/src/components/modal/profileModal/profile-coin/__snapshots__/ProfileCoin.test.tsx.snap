// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile coin test Full page snapshot 1`] = `
<div
  class="coinItem"
  data-testid="profile-coin"
>
  <div>
    <div>
      <img
        alt="Not found"
        src="https://assets.coincap.io/assets/icons/btc@2x.png"
      />
      <span>
        BTC
      </span>
    </div>
    <span>
      Price:
       
      $20,000.00
    </span>
    <span>
      Count: 
      3
    </span>
  </div>
  <button>
    remove
  </button>
</div>
`;
